openupgrade_scripts/scripts/loyalty/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,815 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_field_renames = [
+    ("coupon.program", "coupon_program", "promo_applicability", "applies_on"),
+    ("coupon.program", "coupon_program", "promo_code_usage", "trigger"),
+    ("coupon.program", "coupon_program", "maximun_use_number", "max_usage"),
+    ("coupon.reward", "coupon_reward", "reward_description", "description"),
+    ("coupon.reward", "coupon_reward", "discount_apply_on", "discount_applicability"),
+    ("coupon.reward", "coupon_reward", "discount_type", "discount_mode"),
+    (
+        "coupon.reward",
+        "coupon_reward",
+        "discount_specific_product_ids",
+        "discount_product_ids",
+    ),
+    ("coupon.reward", "coupon_reward", "reward_product_quantity", "reward_product_qty"),
+    ("coupon.rule", "coupon_rule", "rule_minimum_amount", "minimum_amount"),
+    ("coupon.rule", "coupon_rule", "rule_min_quantity", "minimum_qty"),
+    ("coupon.rule", "coupon_rule", "rule_products_domain", "product_domain"),
+    (
+        "coupon.rule",
+        "coupon_rule",
+        "minimum_amount_tax_inclusion",
+        "minimum_amount_tax_mode",
+    ),
+]
+_models_renames = [
+    ("coupon.program", "loyalty.program"),
+    ("coupon.reward", "loyalty.reward"),
+    ("coupon.coupon", "loyalty.card"),
+    ("coupon.rule", "loyalty.rule"),
+]
+_tables_renames = [
+    ("coupon_program", "loyalty_program"),
+    ("coupon_reward", "loyalty_reward"),
+    ("coupon_coupon", "loyalty_card"),
+    ("coupon_rule", "loyalty_rule"),
+]
+_xmlids_renames = [
+    (
+        "loyalty.coupon_action",
+        "loyalty.loyalty_card_action",
+    ),
+    (
+        "loyalty.coupon_generate",
+        "loyalty.loyalty_generate_wizard_action",
+    ),
+    (
+        "loyalty.report_coupon_code",
+        "loyalty.report_loyalty_card",
+    ),
+    (
+        "loyalty.coupon_view_form",
+        "loyalty.loyalty_card_view_form",
+    ),
+    (
+        "loyalty.coupon_view_tree",
+        "loyalty.loyalty_card_view_tree",
+    ),
+    (
+        "loyalty.coupon_generate_view_form",
+        "loyalty.loyalty_generate_wizard_view_form",
+    ),
+    (
+        "loyalty.coupon_program_view_form_common",
+        "loyalty.loyalty_program_view_form",
+    ),
+    (
+        "loyalty.coupon_program_view_search",
+        "loyalty.loyalty_program_view_search",
+    ),
+    (
+        "loyalty.coupon_program_view_tree",
+        "loyalty.loyalty_program_view_tree",
+    ),
+    (
+        "sale_loyalty.mail_template_sale_coupon",
+        "loyalty.mail_template_loyalty_card",
+    ),
+    (
+        "gift_card.gift_card_product_50",
+        "loyalty.gift_card_product_50",
+    ),
+]
+
+
+def update_loyalty_program_data(env):
+    # Fill date_to fill: This data takes the values of the following fields:
+    # - validity_duration (coupon.program): in case of coupon programs that generate
```

Comment:
Wasn't this already set in v15 :thinking:

Reply:
In v15 the coupon view was set to a validity in days (validity_duration) and in the case of the promotions view a validity date was set in the rules (rule_date_to). In v16 there is no distinction for coupons or promotions, everything goes into loyalty programs and is set in the new date_to field.